### PR TITLE
Fix fragment transitions

### PR DESCRIPTION
**Pull request by huddy987**
Changed several fragments to use hide and show instead of replace to improve the user experience

### CODE REVIEW

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksActivity.java [hunk 1]
```diff
@@ -15,36 +15,82 @@
...
 
 public class MyBooksActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener
 {
-
-    private BottomNavigationView navbar;
     private MyBooksFragment myBooksFragment = new MyBooksFragment();
     private SearchFragment searchFragment = new SearchFragment();
     private RequestsFragment requestsFragment = new RequestsFragment();
     private ProfileFragment profileFragment = new ProfileFragment();
+    private NotificationFragment notificationFragment = new NotificationFragment();
```
Comment (JawadHossain) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksActivity.java:30]: We probably should not create a notification fragment here. Try this:

In MyBooksFragment.java inside onNotificationClick:
        NotificationFragment notificationFragment = new NotificationFragment();
        getFragmentManager().beginTransaction().add(R.id.frame_container, notificationFragment, "notificationFragment").hide(ActiveFragmentTracker.activeFragment).commit();

In NotificationFragment.java inside onBackClick:
        Fragment activeFragment = ActiveFragmentTracker.activeFragment;
        getFragmentManager().beginTransaction().hide(this).show(activeFragment).commit();

In MyBooksActivity.java inside onNavigationItemSelected:
/* If NotificationFragment is open, close it*/
        Fragment notificationFragment = fragmentManager.findFragmentByTag("notificationFragment");
        if( notificationFragment != null){
            fragmentManager.beginTransaction().remove(notificationFragment).commit();
        }

**Review by JawadHossain — CHANGES_REQUESTED** [review]
Amazing work!
Requesting not to create NotificationFragment in MyBooksActivity.java. Pls see the suggestion comment.

**Review by IsaakC — APPROVED** [review]
just a couple hardcoded strings to catch

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksActivity.java (hunk 1 above)
Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksActivity.java:30]: Thanks for the suggestion, fixed

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksFragment.java [hunk 2]
```diff
@@ -276,7 +279,7 @@ private void onFilterClick(View view)
         if (fragmentRootView == null)
         {
             /* Fragment was hidden, show it */
-            getFragmentManager().beginTransaction().add(R.id.frame_container, filterMenu).commit();
+            getFragmentManager().beginTransaction().add(R.id.frame_container, filterMenu, "filterMenu").commit();
```
[the remark this answers is not in the document]
Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/MyBooksFragment.java]: fixed

**Review by JawadHossain — APPROVED** [review]

**Pull request merged**